src/modules/site-v2/base/views/tools/genetic_mapping.py

```diff
@@ -107,10 +107,10 @@ def submit():
 
   # Upload input file to server temporarily, and start the job
   try:
-    with LocalFile(request.files.get('file'), valid_file_extensions=TOOL_INPUT_DATA_VALID_FILE_EXTENSIONS) as filepath:
+    with LocalFile(request.files.get('file'), valid_file_extensions=TOOL_INPUT_DATA_VALID_FILE_EXTENSIONS) as file:

```

Comment:
Now passing the `LocalFile` object to the submission code directly, which helps distinguish between:
- a string that should be uploaded as a file
- a local file object that should be uploaded as-is